lib/gitWorkflow.js

```diff
@@ -108,56 +108,71 @@ class GitWorkflow {
   }
 
   /**
-   * List and delete untracked files
+   * Get a list of all files with both staged and unstaged modifications.
+   * Renames have special treatment, since the single status line includes
+   * both the "from" and "to" filenames, where "from" is no longer on disk.
    */
-  async cleanUntrackedFiles() {
-    const lsFiles = await this.execGit(['ls-files', '--others', '--exclude-standard'])
-    const untrackedFiles = lsFiles
+  async getPartiallyStagedFiles() {
+    debug('Getting partially staged files...')
+    const status = await this.execGit(['status', '--porcelain'])
+    const partiallyStaged = status
       .split('\n')
-      .filter(Boolean)
-      .map(file => path.resolve(this.gitDir, file))
-    await Promise.all(untrackedFiles.map(file => unlink(file)))
+      .filter(line => {
+        /**
+         * See https://git-scm.com/docs/git-status#_short_format
+         * The first letter of the line represents current index status,
+         * and second the working tree
+         */
+        const [index, workingTree] = line
+        return index !== ' ' && workingTree !== ' ' && index !== '?' && workingTree !== '?'
+      })
+      .map(line => line.substr(3)) // Remove first three letters (index, workingTree, and a whitespace)
+      .reduce((fixed, file) => {
+        // if file has been renamed, split it to two
+        const RENAME = ' -> '
+        return fixed.concat(file.includes(RENAME) ? file.split(RENAME) : file)
```

Comment:
This was a bit tricky, in case of renamed files the entry reads like `xx FROM -> TO`, where `FROM` and `TO` are two different files. I needed reduce to get them both into the array.

Reply:
Now let's pray they won't change this in future :)

Reply:
I’ve actually been thinking of using [isomorphic-git](https://isomorphic-git.org/) instead of spawning in cases like this. It can handle everything but stashing and diffing.

I have a working branch, but haven’t updated the tests.

Thoughts?

Reply:
I think it makes sense to me, although I'm not sure about the implications and package size.

What would this give us?

Reply:
It would certainly have more dependencies, but since it is a native JS implementation, it would enable us to skip spawning the external git cli, and it would work the same way regardless of OS or git version.

I’ll have to think about it some more. Since not everything is possible, it might not be worth it.

Reply:
Fixed this with abstraction, and let's leave isomorphic-git for later.